### DISCUSSION TITLE
Sunbird as secure content management system?

### DISCUSSION THREAD

**Discussion by vishaldeltech** [Sunbird-Ed/Community#313] [General Discussion]
Hello

We have a requirement for a CMS and are considering Sunbird for the same as it has robust content storage, approval, and search capabilities. Does Sunbird have the following out of the box:

- Is it possible to have all content locked behind credentials so only logged-in users can see the uploaded content? 
- Is it possible to download every type of content from sunbird to a local hard drive (only logged-in users can do this)? 
- Some of the video files are GBs in size. Will Sunbird support them?

If we need to change the configuration for the above, then any guidance would be appreciated

**Comment by Krishnaj20** [comment]
@vishaldeltech -
Is it possible to have all content locked behind credentials so only logged-in users can see the uploaded content?
_I believe this can be handled in the front end. For ex., if you want to show content only for logged in users then you can mandate login before users are taken to the subsequent screens_
Is it possible to download every type of content from sunbird to a local hard drive (only logged-in users can do this)?
_The player provides an option for users to download their content. Which format are you looking for? I quickly checked for videos, pdf & epub and the download option was there in the player_
Some of the video files are GBs in size. Will Sunbird support them?
_Yep, any video file more than 100mb in size will be streamed online._ 

@vrayulu @maheshkumargangula - Can you check these questions and respond from your side as well?

**Reply by vishaldeltech** [reply]
Hi @Krishnaj20 

I really appreciate the prompt response. Here are my comments on each point:

- **Is it possible to have all content locked behind credentials so only logged-in users can see the uploaded content?** -> _I believe this can be handled in the front end. For ex., if you want to show content only for logged in users then you can mandate login before users are taken to the subsequent screens_

Is it possible to restrict the content behind login in a no-code way? In other words, do I need to spend developer effort for this, or can this be done by changing some configuration file?

- **Is it possible to download every type of content from sunbird to a local hard drive (only logged-in users can do this)?** -> _The player provides an option for users to download their content. Which format are you looking for? I quickly checked for videos, pdf & epub and the download option was there in the player_

We are looking for every format sunbird supports and then some more. The intent here is that course creators can download any content from sunbird (irrespective of type and size) and then use the downloaded file anywhere else they want (for example: Upload to a different LMS). Is this possible in a no-code way?

- **Some of the video files are GBs in size. Will Sunbird support them?** -> _Yep, any video file more than 100mb in size will be streamed online._

This size-related question arose as the documentation of sunbird talked about size limits depending on file type. Here is the link for the same: http://docs.sunbird.org/latest/faqs/content_format/. If we have to make changes, will they be no-code?

**Reply by maheshkumargangula** [reply]
@vishaldeltech 

1. There are changes required to restrict the content completely behind login.
2. The API response contains the download or artefact URL for each content to download. We need to adjust the UI to show these details.
3. Sunbird supports the video files in GB's as well. But, the sunbird instance should enable media steaming to play these contents easily.

Cc: @Krishnaj20

**Reply by vishaldeltech** [reply]
Thank you @maheshkumargangula this is very helpful
